sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/implementation/models/AttestationOpenIdMetadataImpl.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.security.attestation.implementation.models;
+
+import com.azure.security.attestation.models.AttestationOpenIdMetadata;
+
+import java.security.InvalidParameterException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+
+public class AttestationOpenIdMetadataImpl implements AttestationOpenIdMetadata {
+    private String jwksUri;
+    private String issuer;
+    private String[] responseTypesSupported;
+    private String[] tokenSigningAlgorithmsSupported;
+    private String[] supportedClaims;
+
+    @Override public String getJsonWebKeySetUrl() {
+        return jwksUri;
+    }
+
+    @Override public String getIssuer() {
+        return issuer;
+    }
+    @Override public String[] getResponseTypesSupported() {
+        return responseTypesSupported.clone();
+    }
+
+    @Override public String[] getTokenSigningAlgorithmsSupported() {
+        return tokenSigningAlgorithmsSupported.clone();
+    }
+    @Override public String[] getSupportedClaims() {
+        return supportedClaims.clone();
+    }
+
+    public static AttestationOpenIdMetadata fromGenerated(Object generated) {
+        if (!(generated instanceof LinkedHashMap)) {
+            throw new InvalidParameterException("generated OpenId metadata must be a hash map.");
+        }
+
+        @SuppressWarnings("unchecked")
+        LinkedHashMap<String, Object> metadataConfig = (LinkedHashMap<String, Object>) generated;
+        AttestationOpenIdMetadataImpl returnValue = new AttestationOpenIdMetadataImpl();
+
+        if (metadataConfig.containsKey("issuer")) {
+            returnValue.issuer = (String) metadataConfig.get("issuer");
+        }
+
+        if (metadataConfig.containsKey("jwks_uri")) {
+            returnValue.jwksUri = (String) metadataConfig.get("jwks_uri");
+        }
+
+        if (metadataConfig.containsKey("response_types_supported")) {
+            Object responseTypesObject = metadataConfig.get("response_types_supported");
+            if (!(responseTypesObject instanceof ArrayList)) {
+                throw new InvalidParameterException("generated ResponseTypes must be an array list.");
+            }
+            @SuppressWarnings("unchecked")
```

Comment:
curius what does this do?

Reply:
It throws an exception if the server ever returned something that wasn't an ArrayList.
CheckStyle/SpotBugs REALLY want me to add checks before casting anything to a generic, so I add the checks whenever I'm going to cast to a generic.

And then I need to add the CheckStyle/SpotBugs suppression because it doesn't recognize that it was actually checked above.

The internet says that the only alternative to suppressing this warning is copying the object one item at a time, which seems REALLY overkill.